From 53b778338c9eaf56237103dd543965b754da6986 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alejandro=20R=20Sede=C3=B1o?= <asedeno@google.com>
Date: Sun, 15 Sep 2024 10:28:11 -0400
Subject: [PATCH 9/9] Drop unused include of cassert
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Alejandro R Sedeño <asedeno@google.com>
---
 src/http_template.cc | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/http_template.cc b/src/http_template.cc
index 4d5ba92..7f8be50 100644
--- a/src/http_template.cc
+++ b/src/http_template.cc
@@ -16,7 +16,6 @@
 //
 #include "grpc_transcoding/http_template.h"
 
-#include <cassert>
 #include <string>
 #include <vector>
 
-- 
2.46.0.662.g92d0881bb0-goog

