diff --git a/absl/debugging/internal/stacktrace_config.h b/absl/debugging/internal/stacktrace_config.h
index c82d4a36..e632cbb9 100644
--- a/absl/debugging/internal/stacktrace_config.h
+++ b/absl/debugging/internal/stacktrace_config.h
@@ -40,7 +40,7 @@
 // Emscripten stacktraces rely on JS. Do not use them in standalone mode.
 #elif defined(__EMSCRIPTEN__) && !defined(STANDALONE_WASM)
 #define ABSL_STACKTRACE_INL_HEADER \
-  "absl/debugging/internal/stacktrace_emscripten-inl.inc"
+  "absl/debugging/internal/stacktrace_unimplemented-inl.inc"
 
 #elif defined(__ANDROID__) && __ANDROID_API__ >= 33
 #ifdef ABSL_HAVE_THREAD_LOCAL
diff --git a/absl/debugging/symbolize.cc b/absl/debugging/symbolize.cc
index 147fda90..f549300b 100644
--- a/absl/debugging/symbolize.cc
+++ b/absl/debugging/symbolize.cc
@@ -36,7 +36,7 @@
 #elif defined(ABSL_INTERNAL_HAVE_SYMBOLIZE_WIN32)
 // The Windows Symbolizer only works if PDB files containing the debug info
 // are available to the program at runtime.
-#include "absl/debugging/symbolize_win32.inc"
+#include "absl/debugging/symbolize_unimplemented.inc"
 #elif defined(__APPLE__)
 #include "absl/debugging/symbolize_darwin.inc"
 #elif defined(ABSL_INTERNAL_HAVE_SYMBOLIZE_WASM)
diff --git a/absl/flags/commandlineflag.h b/absl/flags/commandlineflag.h
index 9336d6d4..0dba5b17 100644
--- a/absl/flags/commandlineflag.h
+++ b/absl/flags/commandlineflag.h
@@ -163,7 +163,7 @@ class CommandLineFlag {
   bool ParseFrom(absl::string_view value, std::string* absl_nonnull error);
 
  protected:
-  ~CommandLineFlag() = default;
+  virtual ~CommandLineFlag() = default;
 
  private:
   friend class flags_internal::PrivateHandleAccessor;
