diff --git a/BUILD b/BUILD
index a855aa5a9..65cfb4267 100644
--- a/BUILD
+++ b/BUILD
@@ -114,7 +114,10 @@ boringssl_copts = select({
     "@platforms//os:linux": ["-D_XOPEN_SOURCE=700"],
     # Without WIN32_LEAN_AND_MEAN, <windows.h> pulls in wincrypt.h, which
     # conflicts with our <openssl/x509.h>.
-    "@platforms//os:windows": ["-DWIN32_LEAN_AND_MEAN"],
+    "@platforms//os:windows": [
+        "-DWIN32_LEAN_AND_MEAN",
+        "-DDBORINGSSL_IMPLEMENTATION",
+    ],
     "//conditions:default": [],
 }) + asm_copts

@@ -138,6 +141,7 @@ cc_library(
         "@platforms//os:windows": ["-defaultlib:advapi32.lib"],
         "//conditions:default": ["-pthread"],
     }),
+    linkstatic = True,
     visibility = ["//visibility:public"],
 )

@@ -147,6 +151,7 @@ cc_library(
     hdrs = ssl_headers,
     copts = boringssl_copts_cxx,
     includes = ["src/include"],
+    linkstatic = True,
     visibility = ["//visibility:public"],
     deps = [
         ":crypto",
