diff --git a/BUILD b/BUILD
index cfa695a44..e748f9b46 100644
--- a/BUILD
+++ b/BUILD
@@ -101,7 +101,10 @@ linux_copts = posix_copts + [
 boringssl_copts = select({
     "@platforms//os:linux": linux_copts,
     "@platforms//os:macos": posix_copts,
-    "@platforms//os:windows": ["-DWIN32_LEAN_AND_MEAN"],
+    "@platforms//os:windows": [
+        "-DWIN32_LEAN_AND_MEAN",
+        "-DBORINGSSL_IMPLEMENTATION",
+    ],
     "//conditions:default": [],
 })
 
@@ -163,6 +166,7 @@ cc_library(
         "@platforms//os:windows": ["-defaultlib:advapi32.lib"],
         "//conditions:default": ["-lpthread"],
     }),
+    linkstatic = True,
     visibility = ["//visibility:public"],
 )
 
@@ -172,6 +176,7 @@ cc_library(
     hdrs = ssl_headers,
     copts = boringssl_copts_cxx,
     includes = ["src/include"],
+    linkstatic = True,
     visibility = ["//visibility:public"],
     deps = [
         ":crypto",
