diff --git a/../../bazel/tools/cpp/cc_configure.bzl b/cc_configure.bzl
index 610f5a7d5..cd99c404d 100644
--- a/../../bazel/tools/cpp/cc_configure.bzl
+++ b/cc_configure.bzl
@@ -1,3 +1,11 @@
+# This file was imported from https://github.com/bazelbuild/bazel at 3d00b2a. We apply a number of
+# local modifications to deal with known issues in Bazel 0.10.0.
+#
+# * https://github.com/bazelbuild/bazel/issues/2840
+# * (and potentially) https://github.com/bazelbuild/bazel/issues/2805
+#
+# See cc_configure.bzl.diff for the changes made in this fork.
+
 # Copyright 2016 The Bazel Authors. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
@@ -16,7 +24,7 @@
 
 load("@bazel_tools//tools/cpp:windows_cc_configure.bzl", "configure_windows_toolchain")
 load("@bazel_tools//tools/cpp:osx_cc_configure.bzl", "configure_osx_toolchain")
-load("@bazel_tools//tools/cpp:unix_cc_configure.bzl", "configure_unix_toolchain")
+load("//bazel:unix_cc_configure.bzl", "configure_unix_toolchain")
 load("@bazel_tools//tools/cpp:lib_cc_configure.bzl", "get_cpu_value")
 
 def cc_autoconf_impl(repository_ctx, overriden_tools = dict()):
@@ -62,6 +70,7 @@ cc_autoconf = repository_rule(
         "CPLUS_INCLUDE_PATH",
         "CUDA_COMPUTE_CAPABILITIES",
         "CUDA_PATH",
+        "CXX",
         "HOMEBREW_RUBY_PATH",
         "NO_WHOLE_ARCHIVE_OPTION",
         "USE_DYNAMIC_CRT",
