diff --git a/common/types/optional_type.h b/common/types/optional_type.h
index dda7f998..7c885d45 100644
--- a/common/types/optional_type.h
+++ b/common/types/optional_type.h
@@ -147,7 +147,7 @@ class OptionalTypeView final : public OpaqueTypeView {
 
   absl::Span<const Type> parameters() const {
     auto parameters = OpaqueTypeView::parameters();
-    ABSL_DCHECK_EQ(parameters.size(), 1);
+    ABSL_DCHECK_EQ(static_cast<int>(parameters.size()), 1);
     return parameters;
   }
 
