diff --git a/eval/public/cel_function_adapter.h b/eval/public/cel_function_adapter.h
index 7bca2ca..3bed9d3 100644
--- a/eval/public/cel_function_adapter.h
+++ b/eval/public/cel_function_adapter.h
@@ -121,7 +121,7 @@ class FunctionAdapter : public CelFunction {
     return registry->Register(std::move(status).value());
   }
 
-#if defined(__clang_major_version__) && __clang_major_version__ >= 8 && !defined(__APPLE__)
+#if 1
   template <int arg_index>
   inline absl::Status RunWrap(absl::Span<const CelValue> arguments,
                               std::tuple<::google::protobuf::Arena*, Arguments...> input,
@@ -177,7 +177,7 @@ class FunctionAdapter : public CelFunction {
                           "Argument number mismatch");
     }
 
-#if defined(__clang_major_version__) && __clang_major_version__ >= 8 && !defined(__APPLE__)
+#if 1
     std::tuple<::google::protobuf::Arena*, Arguments...> input;
     std::get<0>(input) = arena;
     return RunWrap<0>(arguments, input, result, arena);
