diff --git a/bazel/emscripten_toolchain/crosstool.bzl b/bazel/emscripten_toolchain/crosstool.bzl
index ce87ebb..bb45b42 100644
--- a/bazel/emscripten_toolchain/crosstool.bzl
+++ b/bazel/emscripten_toolchain/crosstool.bzl
@@ -92,6 +92,7 @@ def _impl(ctx):
     tool_paths = [
         tool_path(name = "ar", path = emar_script),
         tool_path(name = "cpp", path = "/bin/false"),
+        tool_path(name = "dwp", path = "/bin/false"),
         tool_path(name = "gcc", path = emcc_script),
         tool_path(name = "gcov", path = "/bin/false"),
         tool_path(name = "ld", path = emcc_link_script),
