diff --git a/single-page-app/verify.sh b/single-page-app/verify.sh
index c05e34e..ced1ead 100755
--- a/single-page-app/verify.sh
+++ b/single-page-app/verify.sh
@@ -132,14 +132,14 @@ test_auth () {
         "${curl_args[@]}"

     responds_with_header \
-        "set-cookie: OauthNonce=" \
+        "set-cookie: OauthNonce" \
         "${proxy_scheme}://localhost:${proxy_port}/login" \
         "${curl_args[@]}"

     random=$(openssl rand -hex 8 | head -c -1)
     hmac=$(echo -n "${random}" | openssl dgst -sha256 -hmac "${HMAC_SECRET}" -binary|base64)
     csrf_token=${random}.${hmac}
-    encoded_state=$(echo -n "{\"url\":\"${proxy_scheme}://localhost:${proxy_port}/login\",\"csrf_token\":\"${csrf_token}\"}" | basenc --base64url --wrap=0 | sed 's/=//g')
+    encoded_state=$(echo -n "{\"url\":\"${proxy_scheme}://localhost:${proxy_port}/login\",\"csrf_token\":\"${csrf_token}\",\"flow_id\":\"${random}\"}" | basenc --base64url --wrap=0 | sed 's/=//g')
     code_verifier="Fc1bBwAAAAAVzVsHAAAAABjf6i_Hvf8T2dEuEhPhhDNMlp16az-0dxisL-TzJKaZjOMF8nov_pG377FHmpKcsA"
     code_challenge="YRQaBq_UpkWzfr6JvtNnh7LMfmPVcIKVYdV98ugwmLY"
     run_log "Fetch the myhub authorization page (${name})"
@@ -154,7 +154,7 @@ test_auth () {

     run_log "Return to the app and receive creds (${name})"
     CODE=$(_curl "${curl_args[@]}" --head "http://localhost:${PORT_MYHUB}/authorize?client_id=0123456789&code_challenge=${code_challenge}&code_challenge_method=S256&redirect_uri=${proxy_scheme}%3A%2F%2Flocalhost%3A${proxy_port}%2Fauthorize&response_type=code&scope=user%3Aemail&state=${encoded_state}" | grep Location | cut -d= -f2 | cut -d\& -f1)
-    RESPONSE=$(_curl "${curl_args[@]}" --cookie "OauthNonce=${csrf_token}" --cookie "CodeVerifier=${code_verifier}" --head "${proxy_scheme}://localhost:${proxy_port}/authorize?code=$CODE&state=${encoded_state}")
+    RESPONSE=$(_curl "${curl_args[@]}" --cookie "OauthNonce.${random}=${csrf_token}" --cookie "CodeVerifier.${random}=${code_verifier}" --cookie "OauthNonce=${csrf_token}" --cookie "CodeVerifier=${code_verifier}" --head "${proxy_scheme}://localhost:${proxy_port}/authorize?code=$CODE&state=${encoded_state}")
     echo "$RESPONSE" | grep "HTTP/1.1 302 Found"
     echo "$RESPONSE" | grep "location: ${proxy_scheme}://localhost:${proxy_port}/login"
     echo "$RESPONSE" | grep "set-cookie: OauthHMAC="
@@ -311,7 +311,7 @@ responds_with_header \
     "http://localhost:${PORT_DEV_PROXY}/login"

 responds_with_header \
-    "set-cookie: OauthNonce=" \
+    "set-cookie: OauthNonce" \
     "http://localhost:${PORT_DEV_PROXY}/login"

 run_log "Test production app (Github)"
@@ -324,6 +324,6 @@ responds_with_header \
     "https://localhost:${PORT_PROXY}/login" \
     -k
 responds_with_header \
-    "set-cookie: OauthNonce=" \
+    "set-cookie: OauthNonce" \
     "https://localhost:${PORT_PROXY}/login" \
     -k
