diff --git a/repository_rules.bzl b/repository_rules.bzl
index efdcf49b..7b65907a 100644
--- a/repository_rules.bzl
+++ b/repository_rules.bzl
@@ -184,19 +184,19 @@ def switched_rules_by_language(
     #
     rules["go_proto_library"] = _switch(
         go,
-        "@io_bazel_rules_go//proto:def.bzl",
+        "@rules_go//proto:def.bzl",
     )
     rules["go_grpc_library"] = _switch(
         go,
-        "@io_bazel_rules_go//proto:def.bzl",
+        "@rules_go//proto:def.bzl",
     )
     rules["go_library"] = _switch(
         go,
-        "@io_bazel_rules_go//go:def.bzl",
+        "@rules_go//go:def.bzl",
     )
     rules["go_test"] = _switch(
         go and grpc and gapic and go_test,
-        "@io_bazel_rules_go//go:def.bzl",
+        "@rules_go//go:def.bzl",
     )
     rules["go_gapic_library"] = _switch(
         go and grpc and gapic,
