From 58e2bf9b0da7d79bbe2fe090c1710ab5c59214ef Mon Sep 17 00:00:00 2001
From: Rohit Agrawal <rohit.agrawal@databricks.com>
Date: Wed, 16 Apr 2025 18:36:38 -0700
Subject: [PATCH] Bazel Fixes

---
 BUILD          | 1 -
 src/core/BUILD | 1 -
 2 files changed, 2 deletions(-)

diff --git a/BUILD b/BUILD
index 9ade4e9bc321c..bb3b6743e860c 100644
--- a/BUILD
+++ b/BUILD
@@ -33,7 +33,6 @@ package(
     default_visibility = ["//visibility:public"],
     features = [
         "-parse_headers",
-        "layering_check",
     ],
 )
 
diff --git a/src/core/BUILD b/src/core/BUILD
index f6cb041711d38..974ab189f5d2d 100644
--- a/src/core/BUILD
+++ b/src/core/BUILD
@@ -27,7 +27,6 @@ licenses(["reciprocal"])
 package(
     default_visibility = ["//:__subpackages__"],
     features = [
-        "layering_check",
     ],
 )
 