diff -ur grpc-1.72.0-orig/BUILD grpc-1.72.0/BUILD
--- grpc-1.72.0-orig/BUILD	2025-05-12 23:01:42
+++ grpc-1.72.0/BUILD	2025-05-12 23:01:50
@@ -33,7 +33,7 @@
     default_visibility = ["//visibility:public"],
     features = [
         "-parse_headers",
-        "layering_check",
+        "-layering_check",
     ],
 )
 
diff -ur grpc-1.72.0-orig/src/core/BUILD grpc-1.72.0/src/core/BUILD
--- grpc-1.72.0-orig/src/core/BUILD	2025-05-12 23:01:43
+++ grpc-1.72.0/src/core/BUILD	2025-05-12 23:01:50
@@ -27,7 +27,7 @@
 package(
     default_visibility = ["//:__subpackages__"],
     features = [
-        "layering_check",
+        "-layering_check",
     ],
 )
 
diff -ur grpc-1.72.0-orig/src/proto/grpc/health/v1/BUILD grpc-1.72.0/src/proto/grpc/health/v1/BUILD
--- grpc-1.72.0-orig/src/proto/grpc/health/v1/BUILD	2025-05-12 23:01:43
+++ grpc-1.72.0/src/proto/grpc/health/v1/BUILD	2025-05-12 23:01:56
@@ -44,3 +44,10 @@
         "health.proto",
     ],
 )
+
+# This alias is required by Envoy's build
+alias(
+    name = "_health_proto_only",
+    actual = ":health_proto",
+    visibility = ["//visibility:public"],
+)
diff -ur grpc-1.72.0-orig/third_party/BUILD grpc-1.72.0/third_party/BUILD
--- grpc-1.72.0-orig/third_party/BUILD	2025-05-12 23:01:41
+++ grpc-1.72.0/third_party/BUILD	2025-05-12 23:06:19
@@ -49,7 +49,7 @@
 
 alias(
     name = "madler_zlib",
-    actual = "@zlib//:zlib",
+    actual = "@envoy//bazel/foreign_cc:zlib",
     tags = ["manual"],
 )
 
@@ -80,7 +80,7 @@
 
 alias(
     name = "cares",
-    actual = "@com_github_cares_cares//:ares",
+    actual = "@envoy//bazel/foreign_cc:ares",
     tags = ["manual"],
 )
 
