diff --git a/BUILD b/BUILD
index 6a23c6c354..80ddecf0bf 100644
--- a/BUILD
+++ b/BUILD
@@ -33,7 +33,7 @@ package(
     default_visibility = ["//visibility:public"],
     features = [
         "-parse_headers",
-        "layering_check",
+        "-layering_check",
     ],
 )
 
diff --git a/src/core/BUILD b/src/core/BUILD
index f65a41fbe3..0a1e3dc631 100644
--- a/src/core/BUILD
+++ b/src/core/BUILD
@@ -27,7 +27,7 @@ licenses(["reciprocal"])
 package(
     default_visibility = ["//:__subpackages__"],
     features = [
-        "layering_check",
+        "-layering_check",
     ],
 )
 
diff --git a/src/core/lib/promise/detail/promise_factory.h b/src/core/lib/promise/detail/promise_factory.h
index a4cd1bacc8..8a21719cc7 100644
--- a/src/core/lib/promise/detail/promise_factory.h
+++ b/src/core/lib/promise/detail/promise_factory.h
@@ -188,7 +188,7 @@ class OncePromiseFactory {
   GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION explicit OncePromiseFactory(F f)
       : f_(std::move(f)) {}
 
-  GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION Promise Make(Arg&& a) {
+  GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION inline Promise Make(Arg&& a) {
     return PromiseFactoryImpl(std::move(f_), std::forward<Arg>(a));
   }
 };
diff --git a/third_party/BUILD b/third_party/BUILD
index 77cb52d0fc..0fd5bae3ab 100644
--- a/third_party/BUILD
+++ b/third_party/BUILD
@@ -18,19 +18,19 @@ package(default_visibility = ["//:__subpackages__"])
 
 alias(
     name = "libssl",
-    actual = "@boringssl//:ssl",
+    actual = "@envoy//bazel:boringssl",
     tags = ["manual"],
 )
 
 alias(
     name = "libcrypto",
-    actual = "@boringssl//:crypto",
+    actual = "@envoy//bazel:crypto",
     tags = ["manual"],
 )
 
 alias(
     name = "madler_zlib",
-    actual = "@zlib//:zlib",
+    actual = "@envoy//bazel/foreign_cc:zlib",
     tags = ["manual"],
 )
 
@@ -61,7 +61,7 @@ alias(
 
 alias(
     name = "cares",
-    actual = "@com_github_cares_cares//:ares",
+    actual = "@envoy//bazel/foreign_cc:ares",
     tags = ["manual"],
 )
 
