diff --git a/BUILD b/BUILD
index 6a23c6c354..80ddecf0bf 100644
--- a/BUILD
+++ b/BUILD
@@ -33,7 +33,7 @@ package(
     default_visibility = ["//visibility:public"],
     features = [
         "-parse_headers",
-        "layering_check",
+        "-layering_check",
     ],
 )
 
diff --git a/src/core/BUILD b/src/core/BUILD
index f65a41fbe3..0a1e3dc631 100644
--- a/src/core/BUILD
+++ b/src/core/BUILD
@@ -27,7 +27,7 @@ licenses(["reciprocal"])
 package(
     default_visibility = ["//:__subpackages__"],
     features = [
-        "layering_check",
+        "-layering_check",
     ],
 )
 
diff --git a/third_party/BUILD b/third_party/BUILD
index 77cb52d0fc..1ebc72ada6 100644
--- a/third_party/BUILD
+++ b/third_party/BUILD
@@ -61,7 +61,7 @@ alias(
 
 alias(
     name = "cares",
-    actual = "@com_github_cares_cares//:ares",
+    actual = "@envoy//bazel/foreign_cc:ares",
     tags = ["manual"],
 )
 
