diff --git a/BUILD b/BUILD
index 5caf32799b..4e05babe22 100644
--- a/BUILD
+++ b/BUILD
@@ -33,7 +33,7 @@ package(
     default_visibility = ["//visibility:public"],
     features = [
         "-parse_headers",
-        "layering_check",
+        "-layering_check",
     ],
 )
 
diff --git a/src/core/BUILD b/src/core/BUILD
index dee69ed3bd..e82006c31e 100644
--- a/src/core/BUILD
+++ b/src/core/BUILD
@@ -25,7 +25,7 @@ licenses(["reciprocal"])
 package(
     default_visibility = ["//:__subpackages__"],
     features = [
-        "layering_check",
+        "-layering_check",
     ],
 )
 
diff --git a/third_party/BUILD b/third_party/BUILD
index 77cb52d0fc..5f69f71978 100644
--- a/third_party/BUILD
+++ b/third_party/BUILD
@@ -30,7 +30,7 @@ alias(
 
 alias(
     name = "madler_zlib",
-    actual = "@zlib//:zlib",
+    actual = "@envoy//bazel/foreign_cc:zlib",
     tags = ["manual"],
 )
 
@@ -61,7 +61,7 @@ alias(
 
 alias(
     name = "cares",
-    actual = "@com_github_cares_cares//:ares",
+    actual = "@envoy//bazel/foreign_cc:ares",
     tags = ["manual"],
 )
 
