diff --git a/BUILD b/BUILD
index 1234567..abcdefg 100644
--- a/BUILD
+++ b/BUILD
@@ -22,11 +22,11 @@ cc_library(
         "jwt_verify_lib/verify.h",
     ],
     deps = [
-        "//external:abseil_flat_hash_set",
-        "//external:abseil_strings",
-        "//external:abseil_time",
-        "//external:protobuf",
-        "//external:ssl",
+        "@com_google_absl//absl/container:flat_hash_set",
+        "@com_google_absl//absl/strings",
+        "@com_google_absl//absl/time",
+        "@com_google_protobuf//:protobuf",
+        "@envoy//bazel:boringssl",
     ],
 )
 
@@ -36,7 +36,7 @@ cc_library(
         "simple_lru_cache/simple_lru_cache_inl.h",
     ],
     deps = [
-        "//external:abseil_flat_hash_map",
+        "@com_google_absl//absl/container:flat_hash_map",
     ],
 )
 
@@ -52,7 +52,7 @@ cc_test(
     linkstatic = 1,
     deps = [
         ":jwt_verify_lib",
-        "//external:googletest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
 
@@ -68,7 +68,7 @@ cc_test(
     linkstatic = 1,
     deps = [
         ":jwt_verify_lib",
-        "//external:googletest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
 
@@ -85,7 +85,7 @@ cc_test(
     linkstatic = 1,
     deps = [
         ":jwt_verify_lib",
-        "//external:googletest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
 
@@ -101,7 +101,7 @@ cc_test(
     linkstatic = 1,
     deps = [
         ":simple_lru_cache_lib",
-        "//external:googletest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
 
@@ -118,7 +118,7 @@ cc_test(
     linkstatic = 1,
     deps = [
         ":jwt_verify_lib",
-        "//external:googletest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
 
@@ -135,7 +135,7 @@ cc_test(
     linkstatic = 1,
     deps = [
         ":jwt_verify_lib",
-        "//external:googletest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
 
@@ -152,7 +152,7 @@ cc_test(
     linkstatic = 1,
     deps = [
         ":jwt_verify_lib",
-        "//external:googletest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
 
@@ -169,7 +169,7 @@ cc_test(
     linkstatic = 1,
     deps = [
         ":jwt_verify_lib",
-        "//external:googletest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
 
@@ -186,7 +186,7 @@ cc_test(
     linkstatic = 1,
     deps = [
         ":jwt_verify_lib",
-        "//external:googletest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
 
@@ -203,7 +203,7 @@ cc_test(
     linkstatic = 1,
     deps = [
         ":jwt_verify_lib",
-        "//external:googletest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
 
@@ -220,7 +220,7 @@ cc_test(
     linkstatic = 1,
     deps = [
         ":jwt_verify_lib",
-        "//external:googletest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
 
@@ -237,7 +237,7 @@ cc_test(
     linkstatic = 1,
     deps = [
         ":jwt_verify_lib",
-        "//external:googletest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
 
@@ -254,7 +254,7 @@ cc_test(
     linkstatic = 1,
     deps = [
         ":jwt_verify_lib",
-        "//external:googletest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
 
@@ -271,6 +271,6 @@ cc_test(
     linkstatic = 1,
     deps = [
         ":jwt_verify_lib",
-        "//external:googletest_main",
+        "@com_google_googletest//:gtest_main",
     ],
 )
diff --git a/test/fuzz/BUILD b/test/fuzz/BUILD
index 28247dd..e35142b 100644
--- a/test/fuzz/BUILD
+++ b/test/fuzz/BUILD
@@ -1,5 +1,6 @@
 load("@rules_fuzzing//fuzzing:cc_defs.bzl", "cc_fuzz_test")
-# load cc_proto_library
+load("@com_google_protobuf//bazel:cc_proto_library.bzl", "cc_proto_library")
+load("@com_google_protobuf//bazel:proto_library.bzl", "proto_library")
 
 licenses(["notice"])
 
@@ -27,7 +27,7 @@ cc_fuzz_test(
     deps = [
         ":jwt_verify_lib_fuzz_input_cc_proto",
         "//:jwt_verify_lib",
-        "//external:libprotobuf_mutator",
+        "@com_github_google_libprotobuf_mutator//:libprotobuf_mutator",
     ],
 )
 
@@ -41,7 +41,7 @@ cc_test(
     deps = [
         ":jwt_verify_lib_fuzz_input_cc_proto",
         "//:jwt_verify_lib",
-        "//external:abseil_strings",
-        "//external:googletest_main",
+        "@com_google_absl//absl/strings",
+        "@com_google_googletest//:gtest_main",
     ],
 )
