diff --git a/../../bazel/tools/cpp/osx_cc_configure.bzl b/./osx_cc_configure.bzl
index bac3c0dfc..29b56d14d 100644
--- a/../../bazel/tools/cpp/osx_cc_configure.bzl
+++ b/./osx_cc_configure.bzl
@@ -1,3 +1,12 @@
+# This file was imported from https://github.com/bazelbuild/bazel at 7b85122. We apply a number of
+# local modifications to deal with known issues in Bazel 0.6.0:
+#
+# * https://github.com/bazelbuild/bazel/issues/2840
+# * (and potentially) https://github.com/bazelbuild/bazel/issues/2805
+# * https://github.com/bazelbuild/bazel/issues/3838
+#
+# See osx_cc_configure.bzl.diff for the changes made in this fork.
+
 # pylint: disable=g-bad-file-header
 # Copyright 2016 The Bazel Authors. All rights reserved.
 #
@@ -17,12 +26,12 @@
 load("@bazel_tools//tools/osx:xcode_configure.bzl", "run_xcode_locator")
 
 load(
-    "@bazel_tools//tools/cpp:lib_cc_configure.bzl",
+    "//bazel:lib_cc_configure.bzl",
     "escape_string",
 )
 
 load(
-    "@bazel_tools//tools/cpp:unix_cc_configure.bzl",
+    "//bazel:unix_cc_configure.bzl",
     "get_escaped_cxx_inc_directories",
     "tpl",
     "get_env",
@@ -79,7 +88,7 @@ def configure_osx_toolchain(repository_ctx):
         Label("@bazel_tools//tools/osx/crosstool:wrapped_clang_pp.tpl"),
         "wrapped_clang_pp")
     repository_ctx.symlink(
-        Label("@bazel_tools//tools/osx/crosstool:BUILD.tpl"),
+        Label("//bazel/osx/crosstool:BUILD.tpl"),
         "BUILD")
     repository_ctx.symlink(
         Label("@bazel_tools//tools/osx/crosstool:osx_archs.bzl"),
@@ -92,7 +101,7 @@ def configure_osx_toolchain(repository_ctx):
       escaped_cxx_include_directories.append("# Error: " + xcodeloc_err + "\n")
     repository_ctx.template(
         "CROSSTOOL",
-        Label("@bazel_tools//tools/osx/crosstool:CROSSTOOL.tpl"),
+        Label("//bazel/osx/crosstool:CROSSTOOL.tpl"),
         {"%{cxx_builtin_include_directory}": "\n".join(escaped_cxx_include_directories)})
   else:
     configure_unix_toolchain(repository_ctx, cpu_value = "darwin")
