diff --git a/bazel/pgv_proto_library.bzl b/bazel/pgv_proto_library.bzl
index b8fb865..f1c82bf 100644
--- a/bazel/pgv_proto_library.bzl
+++ b/bazel/pgv_proto_library.bzl
@@ -9,9 +9,9 @@ def pgv_go_proto_library(name, compilers = _DEFAULT_GO_PROTOC, proto = None, de
     go_proto_library(
         name = name,
         proto = proto,
-        deps = ["@com_envoyproxy_protoc_gen_validate//validate:validate"] + deps,
+        deps = ["@protoc-gen-validate//validate:validate"] + deps,
         compilers = compilers + [
-            "@com_envoyproxy_protoc_gen_validate//bazel/go:pgv_plugin_go",
+            "@protoc-gen-validate//bazel/go:pgv_plugin_go",
         ],
         visibility = ["//visibility:public"],
         **kwargs
@@ -22,4 +22,6 @@ def pgv_cc_proto_library(
         deps = [],
         cc_deps = [],
         copts = [],
+        re2 = None,
+        protobuf = None,
         **kargs):
@@ -45,13 +47,13 @@ def pgv_cc_proto_library(
         srcs = [":" + name + "_validate"],
         deps = cc_deps + [
             ":" + name + "_cc_proto",
-            "@com_envoyproxy_protoc_gen_validate//validate:cc_validate",
-            "@com_envoyproxy_protoc_gen_validate//validate:validate_cc",
-            "@com_google_protobuf//:protobuf",
-            "@com_googlesource_code_re2//:re2",
+            "@protoc-gen-validate//validate:cc_validate",
+            "@protoc-gen-validate//validate:validate_cc",
+            protobuf or "@com_google_protobuf//:protobuf",
+            re2 or "@com_googlesource_code_re2//:re2",
         ],
         copts = copts + select({
-            "@com_envoyproxy_protoc_gen_validate//bazel:windows_x86_64": ["-DWIN32"],
+            "@protoc-gen-validate//bazel:windows_x86_64": ["-DWIN32"],
             "//conditions:default": [],
         }),
         alwayslink = 1,
diff --git a/bazel/protobuf.bzl b/bazel/protobuf.bzl
index 1a41f68..4d95b0d 100644
--- a/bazel/protobuf.bzl
+++ b/bazel/protobuf.bzl
@@ -107,7 +107,7 @@ cc_proto_gen_validate = rule(
             providers = [ProtoInfo],
         ),
         "_validate_deps": attr.label_list(
-            default = [Label("@com_googlesource_code_re2//:re2")],
+            default = [Label("@re2")],
         ),
         "_protoc": attr.label(
             cfg = "exec",
