diff --git a/BUILD b/BUILD
index 887e5e384..40e33c1b0 100644
--- a/BUILD
+++ b/BUILD
@@ -21,4 +21,4 @@ exports_files(["LICENSE"])
 # ZLIB configuration
 ################################################################################

-ZLIB_DEPS = ["@zlib//:zlib"]
+ZLIB_DEPS = ["//external:zlib"]

 ################################################################################
 # Protobuf Runtime Library
diff --git a/python/google/protobuf/__init__.py b/python/google/protobuf/__init__.py
index cd0daeaea..197bf25b6 100644
--- a/python/google/protobuf/__init__.py
+++ b/python/google/protobuf/__init__.py
@@ -31,1 +31,1 @@
 # Copyright 2007 Google Inc. All Rights Reserved.

 __version__ = '3.20.0'
+
+if __name__ != '__main__':
+  try:
+    __import__('pkg_resources').declare_namespace(__name__)
+  except ImportError:
+    __path__ = __import__('pkgutil').extend_path(__path__, __name__)
