diff --git a/BUILD b/BUILD
index 69c9bda..d293092 100644
--- a/BUILD
+++ b/BUILD
@@ -88,7 +88,7 @@ cc_library(
         ":headers",
     ] + select({
         "//bazel:crypto_system": [],
-        "//conditions:default": ["@boringssl//:crypto"],
+        "//conditions:default": ["@envoy//bazel:boringcrypto"],
     }),
     alwayslink = 1,
 )
diff --git a/bazel/external/wasmtime.BUILD b/bazel/external/wasmtime.BUILD
index 79a331f..27da86e 100644
--- a/bazel/external/wasmtime.BUILD
+++ b/bazel/external/wasmtime.BUILD
@@ -21,7 +21,7 @@ genrule(
         "crates/c-api/include/wasm.h",
     ],
     outs = [
-        "include/prefixed_wasm.h",
+        "crates/c-api/include/prefixed_wasm.h",
     ],
     cmd = """
         sed -e 's/\\ wasm_/\\ wasmtime_wasm_/g' \
