--- a/BUILD
+++ b/BUILD
@@ -93,7 +93,7 @@
         "@com_google_absl//absl/cleanup",
     ] + select({
         "//bazel:crypto_system": [],
-        "//conditions:default": ["@boringssl//:crypto"],
+        "//conditions:default": ["@envoy//bazel:crypto"],
     }),
     alwayslink = 1,
 )
@@ -158,7 +158,7 @@
     }),
     deps = [
         ":wasm_vm_headers",
-        "@com_github_bytecodealliance_wasm_micro_runtime//:wamr_lib",
+        "@envoy//bazel/foreign_cc:wamr",
     ],
 )
 
@@ -224,7 +224,7 @@
     deps = [
         ":base_lib",
         ":wasm_vm_headers",
-        "@com_github_bytecodealliance_wasmtime//:wasmtime_lib",
+        "@wasmtime//:wasmtime_lib",
     ],
 )
 
