diff --git a/BUILD b/BUILD
index e531ce1..cc4562c 100644
--- a/BUILD
+++ b/BUILD
@@ -92,7 +92,7 @@ cc_library(
         "@com_google_absl//absl/cleanup",
     ] + select({
         "//bazel:crypto_system": [],
-        "//conditions:default": ["@boringssl//:crypto"],
+        "//conditions:default": ["@envoy//bazel:crypto"],
     }),
     alwayslink = 1,
 )
@@ -156,7 +156,7 @@ cc_library(
     }),
     deps = [
         ":wasm_vm_headers",
-        "@com_github_bytecodealliance_wasm_micro_runtime//:wamr_lib",
+        "@envoy//bazel/foreign_cc:wamr",
     ],
 )
 
@@ -241,7 +241,7 @@ cc_library(
     }),
     deps = [
         ":wasm_vm_headers",
-        "@com_github_bytecodealliance_wasmtime//:wasmtime_lib",
+        "@wasmtime//:wasmtime_lib",
     ],
 )
 
