diff --git a/BUILD b/BUILD
index 91792a8..872131c 100644
--- a/BUILD
+++ b/BUILD
@@ -91,7 +91,7 @@ cc_library(
         ":headers",
     ] + select({
         "//bazel:crypto_system": [],
-        "//conditions:default": ["@boringssl//:crypto"],
+        "//conditions:default": ["@envoy//bazel:boringcrypto"],
     }),
     alwayslink = 1,
 )
@@ -155,7 +155,7 @@ cc_library(
     }),
     deps = [
         ":wasm_vm_headers",
-        "@com_github_bytecodealliance_wasm_micro_runtime//:wamr_lib",
+        "@envoy//bazel/foreign_cc:wamr",
     ],
 )
 
@@ -221,7 +221,7 @@ cc_library(
     }),
     deps = [
         ":wasm_vm_headers",
-        "@com_github_bytecodealliance_wasmtime//:wasmtime_lib",
+        "@com_github_wasmtime//:wasmtime_lib",
     ],
 )
 
@@ -280,7 +280,7 @@ cc_library(
     }),
     deps = [
         ":wasm_vm_headers",
-        "@com_github_bytecodealliance_wasmtime//:prefixed_wasmtime_lib",
+        "@com_github_wasmtime//:prefixed_wasmtime_lib",
     ],
 )
 
