diff --git a/java/repositories.bzl b/java/repositories.bzl
index 7e5b939..e8d10b3 100644
--- a/java/repositories.bzl
+++ b/java/repositories.bzl
@@ -88,7 +88,7 @@ def remote_jdk8_repos(name = ""):
     maybe(
         remote_java_repository,
         name = "remote_jdk8_linux_aarch64",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:linux",
             "@platforms//cpu:aarch64",
         ],
@@ -103,7 +103,7 @@ def remote_jdk8_repos(name = ""):
     maybe(
         remote_java_repository,
         name = "remote_jdk8_linux_s390x",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:linux",
             "@platforms//cpu:s390x",
         ],
@@ -117,7 +117,7 @@ def remote_jdk8_repos(name = ""):
     maybe(
         remote_java_repository,
         name = "remote_jdk8_linux",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:linux",
             "@platforms//cpu:x86_64",
         ],
@@ -132,7 +132,7 @@ def remote_jdk8_repos(name = ""):
     maybe(
         remote_java_repository,
         name = "remote_jdk8_macos_aarch64",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:macos",
             "@platforms//cpu:aarch64",
         ],
@@ -146,7 +146,7 @@ def remote_jdk8_repos(name = ""):
     maybe(
         remote_java_repository,
         name = "remote_jdk8_macos",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:macos",
             "@platforms//cpu:x86_64",
         ],
@@ -161,7 +161,7 @@ def remote_jdk8_repos(name = ""):
     maybe(
         remote_java_repository,
         name = "remote_jdk8_windows",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:windows",
             "@platforms//cpu:x86_64",
         ],
@@ -189,7 +189,7 @@ def remote_jdk11_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk11_linux",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:linux",
             "@platforms//cpu:x86_64",
         ],
@@ -205,7 +205,7 @@ def remote_jdk11_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk11_linux_aarch64",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:linux",
             "@platforms//cpu:aarch64",
         ],
@@ -221,7 +221,7 @@ def remote_jdk11_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk11_linux_ppc64le",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:linux",
             "@platforms//cpu:ppc",
         ],
@@ -237,7 +237,7 @@ def remote_jdk11_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk11_linux_s390x",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:linux",
             "@platforms//cpu:s390x",
         ],
@@ -253,7 +253,7 @@ def remote_jdk11_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk11_macos",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:macos",
             "@platforms//cpu:x86_64",
         ],
@@ -269,7 +269,7 @@ def remote_jdk11_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk11_macos_aarch64",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:macos",
             "@platforms//cpu:aarch64",
         ],
@@ -285,7 +285,7 @@ def remote_jdk11_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk11_win",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:windows",
             "@platforms//cpu:x86_64",
         ],
@@ -301,7 +301,7 @@ def remote_jdk11_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk11_win_arm64",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:windows",
             "@platforms//cpu:arm64",
         ],
@@ -318,7 +318,7 @@ def remote_jdk17_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk17_linux",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:linux",
             "@platforms//cpu:x86_64",
         ],
@@ -334,7 +334,7 @@ def remote_jdk17_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk17_linux_aarch64",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:linux",
             "@platforms//cpu:aarch64",
         ],
@@ -350,7 +350,7 @@ def remote_jdk17_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk17_linux_s390x",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:linux",
             "@platforms//cpu:s390x",
         ],
@@ -366,7 +366,7 @@ def remote_jdk17_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk17_linux_ppc64le",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:linux",
             "@platforms//cpu:ppc",
         ],
@@ -382,7 +382,7 @@ def remote_jdk17_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk17_macos",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:macos",
             "@platforms//cpu:x86_64",
         ],
@@ -398,7 +398,7 @@ def remote_jdk17_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk17_macos_aarch64",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:macos",
             "@platforms//cpu:aarch64",
         ],
@@ -413,7 +413,7 @@ def remote_jdk17_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk17_win",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:windows",
             "@platforms//cpu:x86_64",
         ],
@@ -428,7 +428,7 @@ def remote_jdk17_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk17_win_arm64",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:windows",
             "@platforms//cpu:arm64",
         ],
@@ -446,7 +446,7 @@ def remote_jdk20_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk20_linux",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:linux",
             "@platforms//cpu:x86_64",
         ],
@@ -462,7 +462,7 @@ def remote_jdk20_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk20_linux_aarch64",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:linux",
             "@platforms//cpu:aarch64",
         ],
@@ -478,7 +478,7 @@ def remote_jdk20_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk20_macos",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:macos",
             "@platforms//cpu:x86_64",
         ],
@@ -494,7 +494,7 @@ def remote_jdk20_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk20_macos_aarch64",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:macos",
             "@platforms//cpu:aarch64",
         ],
@@ -509,7 +509,7 @@ def remote_jdk20_repos():
     maybe(
         remote_java_repository,
         name = "remotejdk20_win",
-        target_compatible_with = [
+        exec_compatible_with = [
             "@platforms//os:windows",
             "@platforms//cpu:x86_64",
         ],
diff --git a/toolchains/remote_java_repository.bzl b/toolchains/remote_java_repository.bzl
index 86916ec..5521fcf 100644
--- a/toolchains/remote_java_repository.bzl
+++ b/toolchains/remote_java_repository.bzl
@@ -32,20 +32,20 @@ _toolchain_config = repository_rule(
     },
 )
 
-def remote_java_repository(name, version, target_compatible_with = None, prefix = "remotejdk", **kwargs):
+def remote_java_repository(name, version, exec_compatible_with = None, prefix = "remotejdk", **kwargs):
     """Imports a JDK from a http archive and creates runtime toolchain definitions for it.
 
     Register the toolchains defined by this macro via `register_toolchains("@<name>//:all")`, where
     `<name>` is the value of the `name` parameter.
 
-    Toolchain resolution is determined with target_compatible_with
+    Toolchain resolution is determined with exec_compatible_with
     parameter and constrained with --java_runtime_version flag either having value
     of "version" or "{prefix}_{version}" parameters.
 
     Args:
       name: A unique name for this rule.
       version: Version of the JDK imported.
-      target_compatible_with: Target platform constraints (CPU and OS) for this JDK.
+      exec_compatible_with: Target platform constraints (CPU and OS) for this JDK.
       prefix: Optional alternative prefix for configuration flag value used to determine this JDK.
       **kwargs: Refer to http_archive documentation
     """
@@ -77,7 +77,7 @@ alias(
 )
 toolchain(
     name = "toolchain",
-    target_compatible_with = {target_compatible_with},
+    exec_compatible_with = {exec_compatible_with},
     target_settings = [":version_or_prefix_version_setting"],
     toolchain_type = "@bazel_tools//tools/jdk:runtime_toolchain_type",
     toolchain = "{toolchain}",
@@ -85,7 +85,7 @@ toolchain(
 """.format(
             prefix = prefix,
             version = version,
-            target_compatible_with = target_compatible_with,
+            exec_compatible_with = exec_compatible_with,
             toolchain = "@{repo}//:jdk".format(repo = name),
         ),
     )
