diff --git a/tools/bundletool/BUILD b/tools/bundletool/BUILD
index 37674a6d..7537c56b 100644
--- a/tools/bundletool/BUILD
+++ b/tools/bundletool/BUILD
@@ -1,3 +1,5 @@
+load("@rules_python//python:defs.bzl", "py_binary", "py_library", "py_test")
+
 licenses(["notice"])
 
 py_binary(
diff --git a/tools/plisttool/BUILD b/tools/plisttool/BUILD
index 2102ad9b..6c7c47f4 100644
--- a/tools/plisttool/BUILD
+++ b/tools/plisttool/BUILD
@@ -1,3 +1,5 @@
+load("@rules_python//python:defs.bzl", "py_binary", "py_library", "py_test")
+
 licenses(["notice"])
 
 py_binary(
