diff --git a/language-agent/BUILD b/language-agent/BUILD
index 60e9629..e436cdd 100644
--- a/language-agent/BUILD
+++ b/language-agent/BUILD
@@ -16,7 +16,7 @@
 #
 
 load("@rules_proto//proto:defs.bzl", "proto_library")
-load("@rules_cc//cc:defs.bzl", "cc_proto_library")
+load("@com_google_protobuf//bazel:cc_proto_library.bzl", "cc_proto_library")
 load("@com_github_grpc_grpc//bazel:cc_grpc_library.bzl", "cc_grpc_library")
 
 package(default_visibility = ["//visibility:public"])
