diff --git a/bazel/repositories.bzl b/bazel/repositories.bzl
index c143dc01c..78490084e 100644
--- a/bazel/repositories.bzl
+++ b/bazel/repositories.bzl
@@ -839,9 +839,10 @@ def _proxy_wasm_cpp_sdk():
     external_http_archive(name = "proxy_wasm_cpp_sdk")
 
 def _proxy_wasm_cpp_host():
-    external_http_archive(
+    native.new_local_repository(
         name = "proxy_wasm_cpp_host",
         build_file = "@envoy//bazel/external:proxy_wasm_cpp_host.BUILD",
+        path = "/home/mathetake/proxy-wasm-cpp-host",
     )
 
 def _emscripten_toolchain():
