diff --git a/ndk_cc_toolchain_config.bzl b/ndk_cc_toolchain_config.bzl
index f668c3e..ed01d38 100644
--- a/ndk_cc_toolchain_config.bzl
+++ b/ndk_cc_toolchain_config.bzl
@@ -631,11 +631,6 @@ def ndk_cc_toolchain_config(
                     actions = actions.all_link,
                     flags = ["-Wl,-z,noexecstack"],
                 ),
-                flag_set(
-                    actions = actions.all_link,
-                    flags = ["-Wl,-zseparate-code", "-Wl,--no-rosegment"],
-                    features = ["crosstool_linker_lld"],
-                ),
                 # C++ compiles
                 flag_set(
                     actions = actions.all_cpp_compile,
@@ -1381,7 +1376,7 @@ def ndk_cc_toolchain_config(
                 ),
                 flag_set(
                     actions = actions.all_link,
-                    flags = ["-lc", "-lm", "-latomic", "-ldl"],
+                    flags = ["-lc", "-lm", "-ldl"],
                 ),
                 # Inputs and outputs
                 flag_set(
