diff --git a/../../../../bazel/tools/osx/crosstool/BUILD.tpl b/BUILD.tpl
index 8f5376766..b121d0a60 100644
--- a/../../../../bazel/tools/osx/crosstool/BUILD.tpl
+++ b/BUILD.tpl
@@ -1,3 +1,9 @@
+# This file was imported from https://github.com/bazelbuild/bazel at tag 0.5.3. We apply
+# local modifications to deal with known issues in Bazel 0.5.3:
+#
+# https://github.com/bazelbuild/bazel/issues/209
+#
+# See BUILD.tpl.diff for the changes made in this fork.
 package(default_visibility = ["//visibility:public"])
 
 load(":osx_archs.bzl", "OSX_TOOLS_ARCHS")
@@ -39,6 +45,7 @@ cc_toolchain_suite(
           ":make_hashed_objlist.py",
           ":wrapped_clang",
           ":wrapped_ar",
+          ":wrapped_strip",
           ":xcrunwrapper.sh",
         ],
     )
