diff --git a/toolchains/configs/linux/clang/cc/cc_toolchain_config.bzl b/toolchains/configs/linux/clang/cc/cc_toolchain_config.bzl
index 5dbaa86..3c90e3b 100755
--- a/toolchains/configs/linux/clang/cc/cc_toolchain_config.bzl
+++ b/toolchains/configs/linux/clang/cc/cc_toolchain_config.bzl
@@ -386,7 +386,7 @@ def _impl(ctx):
                 ],
                 flag_groups = [
                     flag_group(
-                        flags = ["-gsplit-dwarf"],
+                        flags = ["-gsplit-dwarf", "-g"],
                         expand_if_available = "per_object_debug_info_file",
                     ),
                 ],
diff --git a/toolchains/configs/linux/clang_libcxx/cc/cc_toolchain_config.bzl b/toolchains/configs/linux/clang_libcxx/cc/cc_toolchain_config.bzl
index 5dbaa86..3c90e3b 100755
--- a/toolchains/configs/linux/clang_libcxx/cc/cc_toolchain_config.bzl
+++ b/toolchains/configs/linux/clang_libcxx/cc/cc_toolchain_config.bzl
@@ -386,7 +386,7 @@ def _impl(ctx):
                 ],
                 flag_groups = [
                     flag_group(
-                        flags = ["-gsplit-dwarf"],
+                        flags = ["-gsplit-dwarf", "-g"],
                         expand_if_available = "per_object_debug_info_file",
                     ),
                 ],
diff --git a/toolchains/configs/linux/gcc/cc/cc_toolchain_config.bzl b/toolchains/configs/linux/gcc/cc/cc_toolchain_config.bzl
index 5dbaa86..3c90e3b 100755
--- a/toolchains/configs/linux/gcc/cc/cc_toolchain_config.bzl
+++ b/toolchains/configs/linux/gcc/cc/cc_toolchain_config.bzl
@@ -386,7 +386,7 @@ def _impl(ctx):
                 ],
                 flag_groups = [
                     flag_group(
-                        flags = ["-gsplit-dwarf"],
+                        flags = ["-gsplit-dwarf", "-g"],
                         expand_if_available = "per_object_debug_info_file",
                     ),
                 ],
