# Fix https://github.com/envoyproxy/envoy-setec/issues/1496#issuecomment-2064844217

diff --git a/quiche/quic/core/http/quic_spdy_stream.cc b/quiche/quic/core/http/quic_spdy_stream.cc
index 4a5c2ede2..d69895055 100644
--- a/quiche/quic/core/http/quic_spdy_stream.cc
+++ b/quiche/quic/core/http/quic_spdy_stream.cc
@@ -1865,6 +1865,18 @@ bool QuicSpdyStream::AreHeaderFieldValuesValid(
   return true;
 }

+void QuicSpdyStream::StopReading() {
+  QuicStream::StopReading();
+  if (GetQuicReloadableFlag(
+          quic_stop_reading_also_stops_header_decompression) &&
+      VersionUsesHttp3(transport_version()) && !fin_received() &&
+      spdy_session_->qpack_decoder()) {
+    // Clean up Qpack decoding states.
+    spdy_session_->qpack_decoder()->OnStreamReset(id());
+    qpack_decoded_headers_accumulator_.reset();
+  }
+}
+
 void QuicSpdyStream::OnInvalidHeaders() { Reset(QUIC_BAD_APPLICATION_PAYLOAD); }

 void QuicSpdyStream::CloseReadSide() {
diff --git a/quiche/quic/core/http/quic_spdy_stream.h b/quiche/quic/core/http/quic_spdy_stream.h
index 10c34b10f..5c0cb0128 100644
--- a/quiche/quic/core/http/quic_spdy_stream.h
+++ b/quiche/quic/core/http/quic_spdy_stream.h
@@ -117,6 +117,7 @@ class QUICHE_EXPORT QuicSpdyStream

   // QuicStream implementation
   void OnClose() override;
+  void StopReading() override;

   // Override to maybe close the write side after writing.
   void OnCanWrite() override;
diff --git a/quiche/quic/core/quic_flags_list.h b/quiche/quic/core/quic_flags_list.h
index a38a699ec..58b052015 100644
--- a/quiche/quic/core/quic_flags_list.h
+++ b/quiche/quic/core/quic_flags_list.h
@@ -105,5 +105,7 @@ QUIC_FLAG(quic_reloadable_flag_quic_bbr2_probe_two_rounds, true)
 QUIC_FLAG(quic_reloadable_flag_quic_bbr2_simplify_inflight_hi, true)
 // When true, the BBR4 copt sets the extra_acked window to 20 RTTs and BBR5 sets it to 40 RTTs.
 QUIC_FLAG(quic_reloadable_flag_quic_bbr2_extra_acked_window, true)
+// If true, QUIC stream will not continue decompressing buffer headers after StopReading() called.
+QUIC_FLAG(quic_reloadable_flag_quic_stop_reading_also_stops_header_decompression, true)
 // When true, we will send HTTP/3 setting 0x33 to indicate that we support RFC 9297.
 QUIC_FLAG(quic_reloadable_flag_quic_enable_h3_datagrams, true)
