# to skip clang build
diff --git a/dlb/libdlb/Makefile b/dlb/libdlb/Makefile
index e7fa1fa..0d5b821 100644
--- a/dlb/libdlb/Makefile
+++ b/dlb/libdlb/Makefile
@@ -34,6 +34,7 @@ SRCS :=         \
 OBJS := $(addprefix $(BUILD_DIR)/, $(SRCS:.c=.o))
 DEPS := $(addprefix $(BUILD_DIR)/, $(SRCS:.c=.d))
 LIB  := libdlb.so
+LIB2  := libdlb.a
 
 default: $(LIB) examples cli doc
 
@@ -55,6 +56,16 @@ $(LIB): $(OBJS)
 	@echo Linking $@
 	@$(CC) $^ -shared $(LDFLAGS) -o $@ -Wl,--version-script=version.map
 
+$(LIB2):
+ifeq '' '$(findstring clang,$(CC))'
+$(LIB2): $(OBJS)
+	@echo Linking $@
+	@$(AR) rcs $@ $^
+else
+$(LIB2):
+	@touch libdlb.a
+endif
+
 examples cli: $(LIB)
 	@$(MAKE) -C $@/
 
