--- copts.bzl	2019-06-03 16:42:49.752903373 -0700
+++ opencensus/copts.bzl	2019-06-03 16:43:58.136847278 -0700
@@ -22,25 +22,25 @@
 """
 
 load(
-    "@com_google_absl//absl:copts/GENERATED_copts.bzl",
-    "ABSL_GCC_FLAGS",
-    "ABSL_GCC_TEST_FLAGS",
-    "ABSL_LLVM_FLAGS",
-    "ABSL_LLVM_TEST_FLAGS",
-    "ABSL_MSVC_FLAGS",
-    "ABSL_MSVC_TEST_FLAGS",
+    "@com_google_absl//absl:copts.bzl",
+    "GCC_FLAGS",
+    "GCC_TEST_FLAGS",
+    "LLVM_FLAGS",
+    "LLVM_TEST_FLAGS",
+    "MSVC_FLAGS",
+    "MSVC_TEST_FLAGS",
 )
 
 WERROR = ["-Werror=return-type", "-Werror=switch"]
 
 DEFAULT_COPTS = select({
-    "//opencensus:llvm_compiler": ABSL_LLVM_FLAGS + WERROR,
-    "//opencensus:windows": ABSL_MSVC_FLAGS,
-    "//conditions:default": ABSL_GCC_FLAGS + WERROR,
+    "//opencensus:llvm_compiler": LLVM_FLAGS + WERROR,
+    "//opencensus:windows": MSVC_FLAGS,
+    "//conditions:default": GCC_FLAGS + WERROR,
 })
 
 TEST_COPTS = DEFAULT_COPTS + select({
-    "//opencensus:llvm_compiler": ABSL_LLVM_TEST_FLAGS + WERROR,
-    "//opencensus:windows": ABSL_MSVC_TEST_FLAGS,
-    "//conditions:default": ABSL_GCC_TEST_FLAGS + WERROR,
+    "//opencensus:llvm_compiler": LLVM_TEST_FLAGS + WERROR,
+    "//opencensus:windows": MSVC_TEST_FLAGS,
+    "//conditions:default": GCC_TEST_FLAGS + WERROR,
 })
--- BUILD	2019-06-03 18:30:58.511651926 -0700
+++ opencensus/exporters/trace/stackdriver/BUILD	2019-06-03 18:32:38.107571186 -0700
@@ -28,7 +28,7 @@
     copts = DEFAULT_COPTS,
     visibility = ["//visibility:public"],
     deps = [
-        "//google/devtools/cloudtrace/v2:tracing_proto",
+        "@googleapis//:tracing_proto",
         "//opencensus/common:version",
         "//opencensus/common/internal/grpc:status",
         "//opencensus/common/internal/grpc:with_user_agent",
