diff --git a/mklove/Makefile.base b/mklove/Makefile.base
index f5084cf5..ab80c3f7 100755
--- a/mklove/Makefile.base
+++ b/mklove/Makefile.base
@@ -97,7 +97,7 @@ $(LIBFILENAME): $(OBJS) $(LIBNAME).lds
 
 $(LIBNAME).a:	$(OBJS)
 	@printf "$(MKL_YELLOW)Creating static library $@$(MKL_CLR_RESET)\n"
-	$(AR) rcs$(ARFLAGS) $@ $(OBJS)
+	$(AR) rcs $@ $(OBJS)
 
 ifeq ($(MKL_NO_SELFCONTAINED_STATIC_LIB),y)
 $(LIBNAME)-static.a:
