diff --git a/mklove/Makefile.base b/mklove/Makefile.base
index 91be4391..ee0952af 100755
--- a/mklove/Makefile.base
+++ b/mklove/Makefile.base
@@ -7,6 +7,22 @@ MKL_YELLOW?=	\033[033m
 MKL_BLUE?=	\033[034m
 MKL_CLR_RESET?=	\033[0m
 
+AR ?= ar
+RANLIB ?= ranlib
+
+ifdef EXT_BUILD_ROOT
+    ifneq ($(findstring /,$(AR)),)
+        ifeq ($(filter /%,$(AR)),)
+            AR := $(EXT_BUILD_ROOT)/$(AR)
+        endif
+    endif
+    ifneq ($(findstring /,$(RANLIB)),)
+        ifeq ($(filter /%,$(RANLIB)),)
+            RANLIB := $(EXT_BUILD_ROOT)/$(RANLIB)
+        endif
+    endif
+endif
+
 DEPS=		$(OBJS:%.o=%.d)
 
 # TOPDIR is "TOPDIR/mklove/../" i.e., TOPDIR.
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index bbe63cff..875189d6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -76,7 +76,6 @@ set(
     tinycthread.c
     tinycthread_extra.c
     rdxxhash.c
-    cJSON.c
 )
 
 if(WITH_SSL)
diff --git a/src/Makefile b/src/Makefile
index 0d0635ce..01ab9528 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -46,7 +46,7 @@ SRCS=		rdkafka.c rdkafka_broker.c rdkafka_msg.c rdkafka_topic.c \
 		rdkafka_assignor.c rdkafka_range_assignor.c \
 		rdkafka_roundrobin_assignor.c rdkafka_sticky_assignor.c \
 		rdkafka_feature.c \
-		rdcrc32.c crc32c.c rdmurmur2.c rdfnv1a.c cJSON.c \
+		rdcrc32.c crc32c.c rdmurmur2.c rdfnv1a.c \
 		rdaddr.c rdrand.c rdlist.c \
 		tinycthread.c tinycthread_extra.c \
 		rdlog.c rdstring.c rdkafka_event.c rdkafka_metadata.c \
diff --git a/src/rdkafka.c b/src/rdkafka.c
index 656076df..b723ef92 100644
--- a/src/rdkafka.c
+++ b/src/rdkafka.c
@@ -74,9 +74,6 @@
 #include <sys/timeb.h>
 #endif
 
-#define CJSON_HIDE_SYMBOLS
-#include "cJSON.h"
-
 #if WITH_CURL
 #include "rdhttp.h"
 #endif
@@ -142,8 +139,6 @@ void rd_kafka_set_thread_sysname(const char *fmt, ...) {
 }
 
 static void rd_kafka_global_init0(void) {
-        cJSON_Hooks json_hooks = {.malloc_fn = rd_malloc, .free_fn = rd_free};
-
         mtx_init(&rd_kafka_global_lock, mtx_plain);
 #if ENABLE_DEVEL
         rd_atomic32_init(&rd_kafka_op_cnt, 0);
@@ -156,8 +151,6 @@ static void rd_kafka_global_init0(void) {
         rd_kafka_ssl_init();
 #endif
 
-        cJSON_InitHooks(&json_hooks);
-
 #if WITH_CURL
         rd_http_global_init();
 #endif
