diff -u -r a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2020-11-23 08:59:08.000000000 -0600
+++ b/CMakeLists.txt	2021-01-15 17:15:43.665745800 -0600
@@ -271,7 +271,11 @@
 if(SIZEOF_SSIZE_T STREQUAL "")
   # ssize_t is a signed type in POSIX storing at least -1.
   # Set it to "int" to match the behavior of AC_TYPE_SSIZE_T (autotools).
-  set(ssize_t int)
+  if(WIN32 AND CMAKE_SIZEOF_VOID_P EQUAL 8)
+    set(ssize_t ptrdiff_t)
+  else()
+    set(ssize_t int)
+  endif()
 endif()
 # AC_TYPE_UINT8_T
 # AC_TYPE_UINT16_T
Only in b: CMakeLists.txt.orig
--- a/lib/nghttp2_session.c	2023-06-14 00:17:10.311464189 +0000
+++ b/lib/nghttp2_session.c	2023-06-14 00:18:49.551376483 +0000
@@ -3294,6 +3294,7 @@
         break;
       }
       if (rv < 0) {
+        int rv2 = 0;
         int32_t opened_stream_id = 0;
         uint32_t error_code = NGHTTP2_INTERNAL_ERROR;
 
@@ -3338,19 +3339,18 @@
         }
         if (opened_stream_id) {
           /* careful not to override rv */
-          int rv2;
           rv2 = nghttp2_session_close_stream(session, opened_stream_id,
                                              error_code);
-
-          if (nghttp2_is_fatal(rv2)) {
-            return rv2;
-          }
         }
 
         nghttp2_outbound_item_free(item, mem);
         nghttp2_mem_free(mem, item);
         active_outbound_item_reset(aob, mem);
 
+        if (nghttp2_is_fatal(rv2)) {
+          return rv2;
+        }
+        
         if (rv == NGHTTP2_ERR_HEADER_COMP) {
           /* If header compression error occurred, should terminiate
              connection. */
