diff --git a/configure.ac b/configure.ac
index 04f23e5..2cf871c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -155,7 +155,7 @@ AS_IF([test -n "${ICP_ROOT}"],
 
     AC_CHECK_LIB([qat], [cpaDcCompressData2], [],
         [AC_MSG_ERROR([libqat not found. Install qatlib.])],
-        [-lusdm -lcrypto])
+        [-lusdm -lcrypto_internal])
 
     AC_CHECK_LIB([qat], [icp_adf_get_numDevices],
         [ADF_CFLAGS="-DADF_PCI_API"
@@ -164,7 +164,7 @@ AS_IF([test -n "${ICP_ROOT}"],
             [SAL_CFLAGS="-DSAL_DEV_API"
              AC_SUBST([SAL_CFLAGS])],
             [AC_MSG_ERROR([QAT device API not found])],
-            [-lusdm -lcrypto])])
+            [-lusdm -lcrypto_internal])])
 
     AC_CHECK_LIB([usdm], [qaeMemAllocNUMA], [],
         [AC_MSG_ERROR([libusdm not found. Install qatlib.])])
