diff --git a/src/Makefile b/src/Makefile
index 6b35340..b45e528 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -57,13 +57,13 @@ ifneq ($(ICP_ROOT), )
 			   -lusdm_drv_s -lnuma
 else
 	# In-tree - look for headers in standard locations
-	QAT_INCLUDE_PATH := $(shell if [ -d /usr/local/include/qat ]; then echo /usr/local/include/qat; elif [ -d /usr/include/qat ]; then echo /usr/include/qat; fi)
+	QAT_INCLUDE_PATH ?= $(shell if [ -d /usr/local/include/qat ]; then echo /usr/local/include/qat; elif [ -d /usr/include/qat ]; then echo /usr/include/qat; fi)
 	ifneq ($(QAT_INCLUDE_PATH), )
         QATFLAGS = -I$(QAT_INCLUDE_PATH)
 	else
         $(error QAT headers not found in Standard path, Please install QATLib development package or set ICP_ROOT for out-of-tree driver)
 	endif
-	LDFLAGS  = -lqat -lusdm -lnuma
+	LDFLAGS  += -lqat -lusdm -lnuma
 endif
 
 ifdef ZSTDLIB
@@ -101,17 +101,11 @@ qatseqprod.o: qatseqprod.c
 
 lib: qatseqprod.o
 	$(AR) rc libqatseqprod.a $^
-	$(CC) -shared $^ $(LDFLAGS) -Wl,-soname,libqatseqprod.so.$(VERSION_MAJOR) -o libqatseqprod.so.$(VERSION)
-	ln -sf libqatseqprod.so.$(VERSION) libqatseqprod.so.$(VERSION_MAJOR)
-	ln -sf libqatseqprod.so.$(VERSION_MAJOR) libqatseqprod.so
 
 .PHONY: install
 install: lib
 	[ -e $(DESTDIR)$(LIBDIR) ] || $(INSTALL) -d -m 755 $(DESTDIR)$(LIBDIR)/
 	[ -e $(DESTDIR)$(INCLUDEDIR) ] || $(INSTALL) -d -m 755 $(DESTDIR)$(INCLUDEDIR)/
-	$(INSTALL_PROGRAM) libqatseqprod.so.$(VERSION) $(DESTDIR)$(LIBDIR)
-	ln -sf libqatseqprod.so.$(VERSION) $(DESTDIR)$(LIBDIR)/libqatseqprod.so.$(VERSION_MAJOR)
-	ln -sf libqatseqprod.so.$(VERSION_MAJOR) $(DESTDIR)$(LIBDIR)/libqatseqprod.so
 	$(INSTALL_DATA) libqatseqprod.a $(DESTDIR)$(LIBDIR)
 	$(INSTALL_DATA) qatseqprod.h $(DESTDIR)$(INCLUDEDIR)
 	@echo qatseqprod library successfully installed
