# Compile in bazel sandbox
--- a/ragel/Makefile.in
+++ b/ragel/Makefile.in
@@ -1540,9 +1540,9 @@ uninstall-am: uninstall-binPROGRAMS
 .PRECIOUS: Makefile
 
 
-version.h: Makefile
-	echo '#define VERSION "$(PACKAGE_VERSION)"' > version.h
-	echo '#define PUBDATE "$(PUBDATE)"' >> version.h
+# version.h: Makefile
+#	echo '#define VERSION "$(PACKAGE_VERSION)"' > version.h
+#	echo '#define PUBDATE "$(PUBDATE)"' >> version.h
 
 @BUILD_PARSERS_TRUE@rlparse.h: rlparse.kh
 @BUILD_PARSERS_TRUE@	kelbt -o $@ $<
