diff --git a/trees.c b/trees.c
index 50cf4b4..e705576 100644
--- a/trees.c
+++ b/trees.c
@@ -870,7 +870,9 @@ void ZLIB_INTERNAL _tr_stored_block(s, buf, stored_len, last)
     bi_windup(s);        /* align on byte boundary */
     put_short(s, (ush)stored_len);
     put_short(s, (ush)~stored_len);
-    zmemcpy(s->pending_buf + s->pending, (Bytef *)buf, stored_len);
+    if (stored_len > 0) {
+        zmemcpy(s->pending_buf + s->pending, (Bytef *)buf, stored_len);
+    }
     s->pending += stored_len;
 #ifdef ZLIB_DEBUG
     s->compressed_len = (s->compressed_len + 3 + 7) & (ulg)~7L;
-- 
2.21.0.593.g511ec345e18-goog

