--- a/templates/cc/register.go	2023-06-22 14:25:05.776175085 +0000
+++ b/templates/cc/register.go	2023-06-22 14:26:33.008090583 +0000
@@ -116,6 +116,10 @@
 func (fns CCFuncs) methodName(name interface{}) string {
 	nameStr := fmt.Sprintf("%s", name)
 	switch nameStr {
+	case "concept":
+		return "concept_"
+	case "requires":
+		return "requires_"
 	case "const":
 		return "const_"
 	case "inline":
diff --git a/validate/BUILD b/validate/BUILD
index a9d38c5..2baa5d2 100644
--- a/validate/BUILD
+++ b/validate/BUILD
@@ -1,9 +1,10 @@
 load("@com_google_protobuf//:protobuf.bzl", "py_proto_library")
+load("@com_google_protobuf//bazel:cc_proto_library.bzl", "cc_proto_library")
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
 load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
-load("@rules_cc//cc:defs.bzl", "cc_library", "cc_proto_library")
+load("@rules_cc//cc:defs.bzl", "cc_library")
 load("@rules_java//java:defs.bzl", "java_proto_library")
 load("@rules_proto//proto:defs.bzl", "proto_library")
-load("@io_bazel_rules_go//go:def.bzl", "go_library")
 
 package(
     default_visibility =
