diff --git a/xds/type/matcher/v3/matcher.proto b/xds/type/matcher/v3/matcher.proto
index da7c1f9..1606e94 100644
--- a/xds/type/matcher/v3/matcher.proto
+++ b/xds/type/matcher/v3/matcher.proto
@@ -118,6 +118,11 @@ message Matcher {
 
       // Extension for custom matching logic.
       core.v3.TypedExtensionConfig custom_match = 4;
+
+      // Longest matching prefix wins. In the event that the sub-matcher within that matcher
+      // doesn't find a match, will try the next longest matching prefix, repeatedly, until
+      // either a match is found or there are no further prefix matches in the map.
+      MatchMap prefix_match_map_with_retry = 5;
     }
   }
 
